File: src/DotNetPad/DotNetPad.Applications/Samples/Tuples.vb
Imports System.Collections.Generic
Imports System.Console

Namespace Waf.DotNetPad.Samples
    Module Tuples
        Sub Main()
            ' Create a tuple with semantic names
            Dim namedLetters1 As (alpha As String, beta As String) = ("a", "b")
            Dim namedLetters2 = (alpha:="a", beta:="b")
            WriteLine($"{namedLetters1.alpha} == {namedLetters2.alpha}")

            ' Use method that returns a tuple
            Dim rangeValue = Range(New List(Of Integer) From {8, 2, 4})
            WriteLine($"Min: {rangeValue.min}; Max: {rangeValue.max}")
        End Sub

        Function Range(numbers As IEnumerable(Of Integer)) As (min As Integer, max As Integer)
            Dim min = Integer.MaxValue
            Dim max = Integer.MinValue
            For Each n In numbers
                min = If(n < min, n, min)
                max = If(n > max, n, max)
            Next
            Range = (min, max)			
        End Function
    End Module
End Namespace

File: src/DotNetPad/DotNetPad.Applications/Samples/Tuples.cs
using System;
using System.Collections.Generic;
using static System.Console;

namespace Waf.DotNetPad.Samples
{
    internal static class Tuples
    {
        internal static void Main()
        {
            // Create a tuple with semantic names
            (string alpha, string beta) namedLetters1 = ("a", "b");
            var namedLetters2 = (alpha: "a", beta: "b");
            WriteLine($"{namedLetters1.alpha} == {namedLetters2.alpha}");

            // Use method that returns a tuple
            var range = Range(new[] { 8, 2, 4 });
            WriteLine($"Min: {range.min}; Max: {range.max}");
            (int min, int max) = Range(new[] { 8, 2, 4 });
            WriteLine($"Min: {min}; Max: {max}");

            // Use deconstruct
            var point1 = new Point(1.1, 1.2);
            (double a, double b) = point1;
            WriteLine($"X: {a}; Y: {b}");
        }

        private static (int min, int max) Range(IEnumerable<int> numbers)
        {
            int min = int.MaxValue;
            int max = int.MinValue;
            foreach (var n in numbers)
            {
                min = (n < min) ? n : min;
                max = (n > max) ? n : max;
            }
            return (min, max);
        }

        public struct Point
        {
            public Point(double x, double y)
            {
                X = x;
                Y = y;
            }

            public double X { get; }
            public double Y { get; }

            public void Deconstruct(out double x, out double y)
            {
                x = X;
                y = Y;
            }
        }
    }
}